Gulpfile.js

```diff
@@ -42,527 +13,6 @@ ll
-        var buildTask = isTravis ? 'build-website' : 'build-website-testing';
-
-        runSequence(buildTask, 'serve-website', resolve);
-    })
-        .then(function () {
-            var WebsiteTester = require('./test/website/test.js');
-            var websiteTester = new WebsiteTester();
-
-            return websiteTester.checkLinks();
-        })
-        .then(function (failed) {
-            return new Promise(function (resolve, reject) {
-                websiteServer.close(function () {
-                    if (failed)
-                        reject('Broken links found!');
-                    else
-                        resolve();
-                });
-            });
-        });
-}
-
-gulp.task('test-website', function () {
-    return testWebsite(false);
-});
-
-gulp.task('test-website-travis', function () {
-    return testWebsite(true);
-});
-
-gulp.task('publish-website', ['build-website-production'], function () {
-    // NOTE: it's accidentally stopped being compatible with node 0.10 without
-    // major version bump due to https://github.com/floridoo/gulp-sourcemaps/issues/236,
-    // so we require it here.
-    var ghpages = require('gulp-gh-pages');
-
-    return gulp
-        .src('site/deploy/**/*')
-        .pipe(rename(function (filePath) {
-            filePath.dirname = filePath.dirname.toLowerCase();
-
-            return filePath;
-        }))
-        .pipe(prompt.confirm({
-            message: 'Are you sure you want to publish the website?',
-            default: false
-        }))
-        .pipe(ghpages());
-});
-
-gulp.task('test-docs-travis', ['test-website-travis', 'lint']);
-
-
-function testFunctional (fixturesDir, testingEnvironmentName) {
-    process.env.TESTING_ENVIRONMENT = testingEnvironmentName;
-
-    return gulp
-        .src(['test/functional/setup.js', fixturesDir + '/**/test.js'])
-        .pipe(mocha({
-            ui:       'bdd',
-            reporter: 'spec',
-            timeout:  typeof v8debug === 'undefined' ? 30000 : Infinity // NOTE: disable timeouts in debug
-        }));
-}
-
-gulp.task('test-functional-travis-desktop-osx-and-ms-edge', ['build'], function () {
-    return testFunctional('test/functional/fixtures', functionalTestConfig.testingEnvironmentNames.saucelabsOSXDesktopAndMSEdgeBrowsers);
-});
-
-gulp.task('test-functional-travis-mobile', ['build'], function () {
-    return testFunctional('test/functional/fixtures', functionalTestConfig.testingEnvironmentNames.saucelabsMobileBrowsers);
-});
-
-gulp.task('test-functional-local', ['build'], function () {
-    return testFunctional('test/functional/fixtures', functionalTestConfig.testingEnvironmentNames.localBrowsers);
-});
-
-gulp.task('test-functional-travis-legacy', ['build'], function () {
-    return testFunctional('test/functional/legacy-fixtures', functionalTestConfig.testingEnvironmentNames.legacy);
-});
+// require all tasks inside ./task directory
+requireAll(path.join(__dirname, '/tasks/'));
```

Comment:
if the all tasks run without bugs, we can finish this PR, and begin to see in details a task, or group of tasks, to improve it